File: festiCarFront/src/models/OffreCovoiturage.ts
import { Etape } from "./Etape";
import { Festival } from "./Festival";
import { Utilisateur } from "./Utilisateur";

export interface OffreCovoirage {
  idOffreCovoiturage: number;
  nbPlace: number;
  numImmatriculation: string;
  modele: string;
  couleur: string;
  conducteur: any;
  festival: any;
  etape: Etape[];
}

File: Back/src/main/java/fr/uga/miage/m1/dto/OffreCovoiturage.java
package fr.uga.miage.m1.dto;

import java.util.List;

import fr.uga.miage.m1.models.ConducteurEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OffreCovoiturage {
    
     long idOffreCovoiturage;

     int nbPlace;

     String numImmatriculation;

     String modele;

     String couleur;

     ConducteurEntity conducteur;

     Festival festival;

     List<Etape> etape;

}
